xarray/core/dataarray.py

```diff
@@ -2853,6 +2861,135 @@ def dot(
 
         return computation.dot(self, other, dims=dims)
 
+    def cov(self, other: "DataArray", dim=None) -> "DataArray":
```

Comment:
We can remove these in favor of the `xr.cov` functions, i.e. we don't need both. Sorry if this was unclear before and that lost you some time

Reply:
OK, I just imitated the dot implementation enterily and thought that it could be a useful method.

Why making a difference between dot and cov? 

I invested some time but learned from it, so no time lost! 🙂